packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -133,415 +133,340 @@ const testSchema: SchemaData = {
-						benchmarkFn: () => {
-							manipulateCursorTree(tree, path, dataType);
-						},
-					});
-				}
-				for (const [numberOfNodes, benchmarkType] of nodesCountWide) {
-					let tree: ISharedTree;
-					let provider: ITestTreeProvider;
-					let path: UpPath;
-					benchmark({
-						type: benchmarkType,
-						title: `Update value at leaf of ${numberOfNodes} wide tree`,
-						before: async () => {
-							provider = await TestTreeProvider.create(1);
-							tree = provider.trees[0];
-							tree.storedSchema.update(testSchema);
-							await insertNodesToTestTree(
-								provider,
-								tree,
-								numberOfNodes,
-								TreeShape.Wide,
-								dataType,
-							);
-							path = getCursorLeafNode(numberOfNodes, TreeShape.Wide);
-						},
-						benchmarkFn: () => {
-							manipulateCursorTree(tree, path, dataType);
-						},
-					});
-				}
-			});
-		}
+		});
 	});
 	describe("EditableTree bench", () => {
-		for (const dataType of testSubtrees.keys()) {
+		for (const [numberOfNodes, benchmarkType] of nodesCountDeep) {
+			let provider: ITestTreeProvider;
+			let trees: readonly ISharedTree[];
+			let tree: ISharedTree;
+			benchmark({
+				type: benchmarkType,
+				title: `Deep Tree with Editable Tree: reads with ${numberOfNodes} nodes`,
+				before: async () => {
+					[provider, trees] = await createSharedTrees(
+						getTestSchema(FieldKinds.optional),
+						[{ type: rootSchemaName, value: 1 }],
+						1,
+					);
+					tree = trees[0];
+					insertNodesToEditableTree(tree, numberOfNodes, TreeShape.Deep);
+				},
+				benchmarkFn: () => {
+					assert.equal(
+						numberOfNodes,
+						readEditableTree(tree, numberOfNodes, TreeShape.Deep),
+					);
+				},
+			});
+		}
+		for (const [numberOfNodes, benchmarkType] of nodesCountWide) {
+			let provider: ITestTreeProvider;
+			let trees: readonly ISharedTree[];
+			let tree: ISharedTree;
+			benchmark({
+				type: benchmarkType,
+				title: `Wide Tree with Editable Tree: reads with ${numberOfNodes} nodes`,
+				before: async () => {
+					[provider, trees] = await createSharedTrees(
+						getTestSchema(FieldKinds.sequence),
+						[{ type: rootSchemaName, value: 1 }],
+						1,
+					);
+					tree = trees[0];
+					insertNodesToEditableTree(tree, numberOfNodes, TreeShape.Wide);
+				},
+				benchmarkFn: () => {
+					assert.equal(
+						numberOfNodes,
+						readEditableTree(tree, numberOfNodes, TreeShape.Wide),
+					);
+				},
+			});
+		}
+		for (const [numberOfNodes, benchmarkType] of nodesCountDeep) {
+			let provider: ITestTreeProvider;
+			let trees: readonly ISharedTree[];
+			let tree: ISharedTree;
+			benchmark({
+				type: benchmarkType,
+				title: `Deep Tree with Editable Tree: writes ${numberOfNodes} nodes`,
+				before: async () => {
+					[provider, trees] = await createSharedTrees(
+						getTestSchema(FieldKinds.sequence),
+						[{ type: rootSchemaName, value: 1 }],
+						1,
+					);
+					tree = trees[0];
+				},
+				benchmarkFn: async () => {
```

Comment:
I don't think this needs to be async. Non async benchmarks have a lot lower measurement overhead, so removing the async here is valuable if possible.

There are also several other functions which I think can be made non async.

Reply:
removed unnecessary async calls